corehq/apps/auditcare/tasks.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import sys
+
+from celery.schedules import crontab
+from celery.task import periodic_task
+from couchdbkit.ext.django.loading import get_db
+
+from dimagi.utils.couch.database import iter_docs
+from dimagi.utils.dates import force_to_datetime
+from dimagi.utils.logging import notify_exception
+
+from corehq.apps.auditcare.models import (
+    ACCESS_LOGIN,
+    ACCESS_LOGOUT,
+    ACCESS_FAILED,
+    AccessAudit,
+    NavigationEventAudit,
+)
+from corehq.apps.auditcare.utils.export import get_sql_start_date
+from corehq.util.soft_assert import soft_assert
+
+log = logging.getLogger(__name__)
+
+ACCESS_LOOKUP = {
+    "login": ACCESS_LOGIN,
+    "logout": ACCESS_LOGOUT,
+    "login_failed": ACCESS_FAILED,
+}
+
+
+@periodic_task(run_every=crontab(minute="30"), queue='background_queue')
```

Comment:
I'm planning to do some sample runs in prod and then tweak this, since 1000 records/hour will get us basically nowhere.

This being a periodic task and failing hard does mean that if it encounters an error, it'll fail continuously until a fix is deployed. That's kind of gross, but also one task starting up and immediately failing each hour isn't a big deal.

I went with a periodic task so that it won't need manual restart if its celery worker goes down. open to other suggestions, again trying to minimize babysitting.

Reply:
What do you think about having this start as a management command that runs X records and prints out verbose debugging information so that for the first few attempts you can babysit it and look at what it's copying in detail and verify that the new SQL records look the way you expect, etc. The process would look something like this

1. Run `django-manage copy_auditcare_events_to_sql 1` to copy a single record that you can inspect carefully
2. Run `django-manage copy_auditcare_events_to_sql 10` to show see that it wasn't a fluke, and look at those records as well
3. Run `django-manage copy_auditcare_events_to_sql 100` and make sure that the checkpointing is working as planned and that none of the original 11 records from before are copied
4. Run `django-manage copy_auditcare_events_to_sql 1000` to get a sense of how fast it runs
5. Run `django-manage copy_auditcare_events_to_sql 10000` to get a sense of what a reasonable limit is and whether this has any effect whatsoever on load metrics on the auditcare db
7. Double check that there are exactly 11,111 new sql records (based on the earliest timestamp before step 1). If it doesn't match perfectly, investigate the checkpointing.
6. Then finally make another PR that brings back this periodic task, with a limit setting based on what you saw in steps 4 & 5. (or instead of another PR, you can flag it in this PR to run or not based on some kind of setting you can switch on)

That seems like a lot more steps, but I think it's actually just giving you breathing space to do tests that you would already be doing if anything went remotely wrong with the periodic task. It also prevents a situation where we realize the periodic task is doing something really bad (like creating records in an infinite loop or something) and are powerless to stop it until the next deploy. So overall I expect splitting it out into a manual phase where you can dip your toes in and make sure things work the way you expect and an automated phase where this is running constantly will be overall less work, especially when taking into account risk-adjustment.

I realize that was a wall of text. Thoughts?

Reply:
It's tempting to reply with just a thumbs-up emoji.

More articulately, yes, I agree in practice I'll want to be able to run this manually and easily change the size before scheduling, and agree starting off with a management command is a good way to facilitate that.

Reply:
Added management command in 573b3d21044299a3437c5990a73b4bb7535f78f0 and removed `periodic_task` decorator for now.